contracts/BaseStrategy.sol

```diff
@@ -584,20 +575,27 @@ abstract contract BaseStrategy {
     function harvest() external onlyKeepers {
         uint256 profit = 0;
         uint256 loss = 0;
+        uint256 debtOutstanding = vault.debtOutstanding();
         uint256 debtPayment = 0;
         if (emergencyExit) {
             // Free up as much capital as possible
-            // NOTE: Don't take performance fee in this scenario
-            (profit, loss, debtPayment) = exitPosition(vault.debtOutstanding());
+            uint256 totalAssets = estimatedTotalAssets();
+            // NOTE: use the larger of total assets or debt oustanding to get profit/loss calculation correct
+            (debtPayment, loss) = liquidatePosition(debtOutstanding > totalAssets ? debtOutstanding : totalAssets);
+            // NOTE: take up any remainder here as profit
+            if (debtPayment > debtOutstanding) {
+                profit = debtPayment.sub(debtOutstanding);
```

Comment:
I don't understand why you take care of profits here and not losses? 
Why not make `liquidatePosition()` only `returns (uint256 _amountFreed)`

Reply:
The assumption is that `liquidatePoisition` should be used to make the Vault whole as quickly as possible, up to `_amountNeeded`. Here we are sending in the maximum amount of assets possible for the Strategy to liquidate.

If there is anything left after paying back the debt, the rest is booked as profits.

Perhaps this is over complicated though? My intention was to use this function in both the `withdraw` and `emergencyShutdown` scenario, because under that scenario we want to withdraw the assets as quickly as possible, slippage be damned.

Reply:
my point is if you have a liquidate position that just returns the amount freed, in `harvest()` you can do:

```
if (debtPayment > debtOutstanding) {
  // profit
} else if (debtPayment < debtOutstanding) {
  // loss
}

Reply:
I think I see your point. Let me see how it looks